app/models/media.rb

```diff
@@ -210,6 +212,17 @@ def get_canonical_url
     nil # Return nil, since url is empty/not a string
   end
 
+  def skip_get_canonical_url?
+    skip_hosts = ['instagram']
+    host = RequestHelper.parse_url(self.original_url).host
+    if host && skip_hosts.any? { |skip_host| host.include?(skip_host) }
```

Comment:
That's a bit too broad. It would match things like `instagram-clone.net`, right? Couldn't we move this logic to the parser classes? The default could be `false`, and then the Instagram parser could implement it as `true`.

Reply:
True! I'll look into making it less generic. 

I think we should move this logic into the parser classes, BUT I'm not sure if that is something that we could do right now, at this point we are still initializing Media. This is something we could look into in a separate ticket.

Something else we should look into is we are parsing the url multiple times, I think more times than we need. We could try to re-write it to parse it once and reuse the result where possible.